### PR TITLE
Use SMC resample function and not default one

### PR DESCRIPTION
**Pull request by emilemathieu**
https://github.com/yebai/Turing.jl/issues/338

### CODE REVIEW

**Comment by yebai** [comment]
Thanks for fixing this!

File: src/samplers/smc.jl [hunk 1]
```diff
@@ -52,7 +52,7 @@ function sample(model::Function, alg::SMC)
   while consume(particles) != Val{:done}
     ess = effectiveSampleSize(particles)
     if ess <= spl.alg.resampler_threshold * length(particles)
-      resample!(particles,use_replay=spl.alg.use_replay)
+      resample!(particles,smc_spl.alg.resampler,use_replay=smc_spl.alg.use_replay)
```
Comment (yebai) [inline src/samplers/smc.jl]: should `smc_spl` be `spl`?

**Comment by emilemathieu** [comment]
Indeed, I copied-paste my PMMH code, ahah how could I messed up a one line PR...

Otherwise @yebai have you seen my other https://github.com/yebai/Turing.jl/issues/338#issuecomment-324557562 ?

**Comment by yebai** [comment]
There is a second reference for `smc_spl`:

```julia
use_replay=smc_spl.alg.use_replay
```
==>

```julia
use_replay=spl.alg.use_replay
```

**Comment by yebai** [comment]
> Otherwise @yebai have you seen my other #338 (comment) ?

Yes, perhaps you can fix that bug in this PR too. Thanks!

**Comment by emilemathieu** [comment]
Fixed

**Pull request merged**